doc/plugins.dox

```diff
@@ -186,59 +195,143 @@ Let's start with the very basic headers and add more later if necessary.
 
-@note
-@a plugindata.h contains the biggest part of the plugin API and provides some basic macros.
-
-Then you should define two basic variables which will give access to data fields
-provided by the plugin API.
-@code
-GeanyPlugin			*geany_plugin;
-GeanyData			*geany_data;
-@endcode
+@note @a plugindata.h contains the biggest part of the plugin API and provides some basic macros.
 
 Now you can go on and write your first lines for your new plugin. As mentioned before,
-you will need to implement and fill out a couple of functions/macros to make the plugin work.
-So let's start with PLUGIN_VERSION_CHECK().
-
-PLUGIN_VERSION_CHECK() is a convenient way to tell Geany which version of Geany's plugin API
-is needed at minimum to run your plugin. The value is defined in
+you will need to implement and a couple of functions. Arguably the most important
+one is @a geany_load_module(). This is the function that makes Geany see you plugin at all. When
+Geany scans the pre-defined and user-configured plugin directories, it will take a look at each
+shared library (or DLL on Windows) to see if it exports a @a geany_load_module() symbol. Files
+lacking these will be outright ignored.
+
+@subsection register Registering a Plugin
+
+Geany will then invoke this function, always and regardless of whether the user activates your
+plugin. In fact its purpose to probe if the plugin should even be presented to the user. Therefore
+it important that you use this function to register your plugin, making it available to the user.
+Geany will pass a pointer to a GeanyPlugin instance which acts as a unique handle to your plugin.
+Use this pointer for registering and later API calls. Registering the plugin consists of a number
+of steps:
+
+  1. Filling GeanyPlugin::info with metadata
+    - @ref PluginInfo::name : The name of your plugin
+    - @ref PluginInfo::description : A brief description.
+    - @ref PluginInfo::version : The current version.
+    - @ref PluginInfo::author : Your contanct information.
+    .
+    Filling all of them is recommended to provide the best user experience, but only the name is truly
+    mandatory.
+  2. Filling GeanyPlugin::funcs with function pointers
+    - @ref GeanyPluginFuncs::init : an initialization function
+    - @ref GeanyPluginFuncs::cleanup : a finalization function
+    - @ref GeanyPluginFuncs::configure : a function that provides configuration (optional)
+    - @ref GeanyPluginFuncs::help : a function that provides help (optional)
+    - @ref GeanyPluginFuncs::callbacks : a pointer to an array of PluginCallback (optional).
+    .
+    @a init and @a cleanup are mandatory, the other ones depend on how advanced your plugin is.
+    Furthermore, the @a init and @a cleanup
+    pair is called when the user activates your plugin in the Plugin Manager. So use these to
+    do advanced initialization and teardown as to not waste resources when the plugin is not even
+    enabled.
```

Comment:
This sounds weird to me, `cleanup` is not called when the plugin is activated.  Also, they are called when the plugin is activated/deactivated, regardless of whether it was from the PM dialog or because it was enabled at startup or still there are shutdown.

What about `@a init and @a cleanup are called when your plugin is activated and deactivated respectively.  Use these to perform further initialization and cleanup not to waste resources when your plugin is not enabled.`